typespec-extension/src/code-model-builder.ts

```diff
@@ -1288,7 +1288,14 @@ export class CodeModelBuilder {
       this.trackSchemaUsage(schema, { serializationFormats: [KnownMediaType.Multipart] });
     }
 
-    if (!schema.language.default.name && schema instanceof ObjectSchema) {
+    if (schema instanceof ObjectSchema && schema.serializationFormats?.includes(KnownMediaType.Multipart)) {
+      // TODO: anonymous model for multipart is not supported
```

Comment:
I don't quite get why is this commit's code change related with 'avoid flatten'?

Reply:
I got a bug on this one. Correct code should be within the `!schema.language.default.name` check.
https://github.com/Azure/autorest.java/blob/05d287353282540d4c4d95b624f1217e0618289f/typespec-extension/src/code-model-builder.ts#L1291-L1306

Reply:
BTW, I probably need to update the anonymous model / flatten logic.

We were using the `Map<String, Object>` to handle that. But this won't work as expected in stream-serialization, as `Map` or `List` would trigger the Jackson flow in core.